common/src/main/java/org/mcaccess/minecraftaccess/features/HUDStatus.java

```diff
@@ -1,19 +1,46 @@
 package org.mcaccess.minecraftaccess.features;
 
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.player.LocalPlayer;
 import net.minecraft.client.resources.language.I18n;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
 
 import org.mcaccess.minecraftaccess.MainClass;
+import org.mcaccess.minecraftaccess.utils.PlayerUtils;
 
 public class HUDStatus {
-    private Boolean wasHidden = Minecraft.getInstance().options.hideGui;
+    private Minecraft client = Minecraft.getInstance();
+    private Boolean hudWasHidden = Minecraft.getInstance().options.hideGui;
+    private boolean attackCooldownPlayed = false;
 
     public void tick() {
-        Boolean isHidden = Minecraft.getInstance().options.hideGui;
+        hudVisibilityStatus();
+        attackCooldownStatus();
+    }
+
+    private void hudVisibilityStatus() {
+        Boolean hudIsHidden = client.options.hideGui;
+
+        if (hudWasHidden != hudIsHidden) {
+            MainClass.narrate(I18n.get(String.format("minecraft_access.hud_status.announce_%s", hudIsHidden ? "hidden" : "shown")), true);
+            hudWasHidden = hudIsHidden;
+        }
+    }
+
+    private void attackCooldownStatus() {
+        boolean indicatorShowing = !hudWasHidden && client.options.attackIndicator().get().getId() != 0 && !PlayerUtils.isSpectator();
```

Comment:
Wouldn't it be clearer to just compare the enum variant instead of getting the ID of the variant and comparing it with a magic number?
```suggestion
        boolean indicatorShowing = !hudWasHidden && client.options.attackIndicator().get() != AttackIndicatorStatus.OFF && !PlayerUtils.isSpectator();
```